common/Cargo.toml

```diff
@@ -33,4 +33,5 @@ proptest.workspace = true
 test-strategy.workspace = true
 
 [build-dependencies]
-vergen = { version = "8.3.2", features = ["cargo", "git", "git2", "rustc" ] }
+vergen = { version = "9.0.6", features = ["cargo", "rustc"] }
+vergen-git2 = "1.0.7"
```

Comment:
We could use `vergen-gix` instead to avoid linking against `libgit2` altogether? But that pulled in a _lot_ more `Cargo.lock` changes, so I just went with the successor to what we had before (vergen 8's `git2` feature).